### PR TITLE
Add new Slime & Vulpkanin names

### PR DESCRIPTION
**Pull request by ParaGitV**
## What Does This PR Do

Changes Slime and Vulpkanin character name generation lists. Previously names were generated using speech verbs which resulted in a lot of similar and unpronounceable names. This adds new Slime/Vulpkanin male/female & last name lists.

## Why It's Good For The Game

Gives randomized character names a wider variety of themed options for their race/gender to hopefully make Slime/Vulp names more memorable than current Slime/Vulp randomized names.

Some examples of our current randomized names below for comparison:
Genagbaram Llbarammaktek
Vuhrurzwichsischt Woortchjarl
Entchauchschzein Cresthzlllnictzch
Zwichswatzchrawr Tchonnghzthvuh
Plopplopboop Popbopboop
Plopplopplop Plopplop
Plopblobblobpop Plopblobploppop
Booppoppopboop Boopboopplopplop

## Images of changes

image

image

## Changelog
:cl:
tweak: Change from syllable-generation to custom lists
/:cl:

### CODE REVIEW

**Comment by atteria** [comment]
I'm sorry, but these are really, really silly.

Just giving Vulps human names erases one of the interesting things they have, which is their language sound. 
>  They tend to sound harsh and guttural, and many a scholar has compared them to the names found in ancient Germanic and Semitic languages. 

And 'Fizzete'? 'Slimeton'? Really? If I saw someone name a character these in-game I would groan.

**Comment by roarbark** [comment]
i like the idea but randomized names for vulpkanin shouldn't be just human names, it takes a lot of flavour out of the race to have vulpkanin predominantly run around with, well just human names.

**Comment by Inadvizable** [comment]
The slime names are meant to be non-pronounceable. Most slimes adapt a name specifically for interacting with other humanoids - and several slime players have canon 'real' names which are those 'popping sound' names.

**Comment by Necaladun** [comment]
People overwhelmingly use these kinds of names anyway, few quick tests showed over 90% of people do at least.

👍

File: code/modules/mob/language.dm [hunk 1]
```diff
@@ -207,6 +207,15 @@
...
...the","Mariah","Mariam","Marilyn","Marina","Marisole","Marivel","Marley","Marni","Marrie","Martina","Mary","Mazelina","Meda","Medea","Mei","Melania","Melanie","Melody","Mercedes","Merle","Meryl","Mia","Michelle","Mila","Millie","Mindy","Miranda","Missy","Misty","Mona","Morgan","Morgana","Morrigan","Morticia","Nadia","Nadine","Nessa","Nia","Nicole","Nikki","Nimah","Nina","Norma","Nova","Olive","Olivia","Opaline","Ophelia","Oriana","Paisley","Paloma","Pam","Pauline","Paz","Penelope","Penny","Phoebe","Piper","Portia","Priya","Rachel","Raina","Raura","Raven","Rayna","Rayne","Rebecca","Regina","Renee","Rhea","Rina","Robin","Rosalind","Rosie","Rowen","Rowena","Royce","Rubella","Ruby","Rue","Ruth","Sabrina","Sadie","Sahara","Sandra","Savina","Sawyer","Selene","Sena","Seraphina","Seraphine","Sheba","Sheila","Sia","Sibylle","Sofie","Sonnet","Sophia","Stacia","Stacy","Stefanie","Stella","Suri","Syden","Sylvia","Tala","Tasha","Tasia","Tatum","Taylee","Teegan","Teresa","Tess","Tessa","Tessy","Theia","Titania","Trisha","Trixie","Trudy","Uma","Ursula","Valenia","Valentina","Vega","Vera","Verena","Vicky","Victoria","Willow","Winnie","Yasmin","Yvette","Yvonne","Zia","Zinnia","Ziva","Zoe","Zuri"))]"
```
Comment (AffectedArc07) [inline code/modules/mob/language.dm:213]: Yeah these are gonna need to be a config.

Take a look at how other names do it with `code\_globalvars\lists\names.dm`

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]

**Comment by Pixelzilla** [comment]
> I'm sorry, but these are really, really silly.
> 
> Just giving Vulps human names erases one of the interesting things they have, which is their language sound.
> 

I can assure you that the current Vulp name generation was not the originally intended naming scheme and was a result of not having a name list in place which basically means it defaults to using the syllables from the Vulp language. Vulp were originally supposed to be a much more open race with a lot of creative freedom for players to do things they wanted instead of trying to shove them all into a very limited box. The current naming/lore is much more restrictive and doesn't allow for as much freedom as the majority of Vulp players clearly want.

Vulp names still have some german-ish options for those who want that but they also have a lot of star/galactic themed names which has always been part of Vulp. If you look at their original hairstyle names they are all galactic/star/constellation themed names. There are also some silly names (just like how humans have silly names like "Butterfill" in their name list). Obviously no one is obligated to change an existing character name or match the new randomized name theme, this is just for people who _wish_ to utilize it when making _new_ characters. It also does not change anything about what their spoken language sounds like.

> The slime names are meant to be non-pronounceable. Most Slimes adapt a name specifically for interacting with other humanoids - and several slime players have canon 'real' names which are those 'popping sound' names.

I am 99% sure that the original slime name generator was not what it currently is. A lot of races that lacked name list/config files defaulted to using their species language syllables. I'm pretty sure this was an unintentional side effect of some other changes and no one really bothered fixing any of the species negatively impacted by it except for Vox and Kidan. If we're gonna argue lore then the current name generation doesn't even fit the naming lore listed on the wiki. Only one of the options listed (pop) is even part of the name generation options and it specifically says most of the names are chosen in galactic common and include variations of certain phrases not that they exclusively use only those words. It even suggests that these alternate names are chosen for **sake of simplicity**. I don't think one can really argue that a name like "Booppoppopboop Boopboopplopplop" is even remotely simple.

File: code/modules/mob/language.dm [hunk 2]
```diff
@@ -207,6 +207,15 @@
 	"hel","ischt","far","wa","baram","iereng","tech","lach","sam","mak","lich","gen","or","ag","eck","gec","stag","onn", \
 	"bin","ket","jarl","vulf","einech","cresthz","azunein","ghzth")
 
+/datum/language/vulpkanin/get_random_name(gender)
+	var/new_name = ""
```
Comment (farie82) [inline code/modules/mob/language.dm:211]:
```suggestion
	var/new_name
```

File: code/modules/mob/language.dm [hunk 3]
```diff
@@ -306,6 +315,15 @@
 	flags = RESTRICTED | WHITELISTED
 	syllables = list("blob","plop","pop","bop","boop")
 
+/datum/language/slime/get_random_name(gender)
+	var/new_name = ""
```
Comment (farie82) [inline code/modules/mob/language.dm:319]:
```suggestion
	var/new_name
```

**Review by farie82 — COMMENTED** [review]
Minor style stuff. You don't have to define a var to be something if you instantly define it afterwards in all cases.

File: config/names/first_female_slime.txt [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+Slusharia
+Slushi
+Slushia
+Slushibell
+Slushie
+Slushy
+Squisha
+Squishie
+Squishlee
+Bloppella
```
Comment (farie82) [inline config/names/first_female_slime.txt:73]: This file has an empty line at the end. While this is normally what we want in DM files, this ain't what we want in config files like this. Here it'll cause DM to see an empty line and parse it as such. So there is a chance that an empty name will be picked for each of these files

**Review by farie82 — CHANGES_REQUESTED** [review]

**Review by farie82 — APPROVED** [review]

**Pull request merged**